### PR TITLE
IBP-4744 enforce variable name and alias uniqueness within a crop

### PR DESCRIPTION
**Pull request by cuenyad**
Please, review.

### CODE REVIEW

**Review by nahuel-soldevilla — APPROVED** [review]

File: src/main/java/org/generationcp/middleware/domain/ontology/VariableOverridesDto.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
+package org.generationcp.middleware.domain.ontology;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+public class VariableOverridesDto {
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/domain/ontology/VariableOverridesDto.java:6]: better to use Pojomatics @cuenyad

File: src/main/java/org/generationcp/middleware/manager/ontology/api/OntologyVariableDataManager.java [hunk 2]
```diff
@@ -106,6 +107,8 @@
 
 	public VariableOverrides getVariableOverridesByVariableIdAndProgram(final Integer variableId, final String programUuid);
 
+	public List<VariableOverridesDto> getVariableOverridesByAliasAndProgram(final String alias, final String programUuid);
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/manager/ontology/api/OntologyVariableDataManager.java:110]: remove finals @cuenyad

File: src/main/java/org/generationcp/middleware/manager/ontology/OntologyVariableDataManagerImpl.java [hunk 3]
```diff
@@ -1031,6 +1032,15 @@ public VariableOverrides getVariableOverridesByVariableIdAndProgram(final Intege
 		}
 	}
 
+	@Override
+	public List<VariableOverridesDto> getVariableOverridesByAliasAndProgram(final String alias, final String programUuid) {
+		try {
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/manager/ontology/OntologyVariableDataManagerImpl.java:1037]: try/catch in DAO should be enough @cuenyad

File: src/main/java/org/generationcp/middleware/manager/ontology/api/OntologyVariableDataManager.java (hunk 2 above)
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/manager/ontology/api/OntologyVariableDataManager.java]: done.

File: src/main/java/org/generationcp/middleware/manager/ontology/OntologyVariableDataManagerImpl.java (hunk 3 above)
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/manager/ontology/OntologyVariableDataManagerImpl.java]: done.

File: src/main/java/org/generationcp/middleware/domain/ontology/VariableOverridesDto.java (hunk 1 above)
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/domain/ontology/VariableOverridesDto.java:7]: done.

**Review by clarysabel — APPROVED** [review]

**Pull request merged**